src/Tizen.NUI/src/internal/Utility/ColorCutQuantizer.cs

```diff
@@ -0,0 +1,633 @@
+                if (Count == Capacity)
+                    Grow();
+                Heap[Tail++] = item;
+                BubbleUp(Tail - 1);
+            }
+
+            public T Peek()
+            {
+                if (Count == 0) throw new InvalidOperationException("CustomHeap is empty");
+                return Heap[0];
+            }
+
+            public T Poll()
+            {
+                if (Count == 0) throw new InvalidOperationException("CustomHeap is empty");
+                T Ret = Heap[0];
+
+                Tail--;
+
+                Swap(Tail, 0);
+
+                BubbleDown(0);
+
+                return Ret;
+            }
+
+            private void BubbleDown(int i)
+            {
+                int DominatingNode = Dominating(i);
+                if (DominatingNode == i) return;
+                Swap(i, DominatingNode);
+                BubbleDown(DominatingNode);
+            }
+
+            private void BubbleUp(int i)
+            {
+                if (i == 0 || Dominates(Heap[Parent(i)], Heap[i]))
+                    return;
+                Swap(i, Parent(i));
+                BubbleUp(Parent(i));
+            }
+
+            private int Dominating(int i)
+            {
+                int DominatingNode = i;
+                DominatingNode = GetDominating(YoungChild(i), DominatingNode);
+                DominatingNode = GetDominating(OldChild(i), DominatingNode);
+                return DominatingNode;
+            }
+
+            private int GetDominating(int newNode, int dominatingNode)
+            {
+                if (newNode < Tail && !Dominates(Heap[dominatingNode], Heap[newNode]))
+                    return newNode;
+                else
+                    return dominatingNode;
+            }
+
+            private void Swap(int i, int j)
+            {
+                T Tmp = Heap[i];
+
+                Heap[i] = Heap[j];
+
+                Heap[j] = Tmp;
+            }
+
+            private static int Parent(int i)
+            {
+                return (i + 1) / 2 - 1;
+            }
+
+            private static int YoungChild(int i)
+            {
+                return (i + 1) * 2 - 1;
+            }
+
+            private static int OldChild(int i)
+            {
+                return YoungChild(i) + 1;
+            }
+
+            private void Grow()
+            {
+                int NewCapacity = Capacity * GrowFactor + MinGrow;
+                var NewHeap = new T[NewCapacity];
+                Array.Copy(Heap, NewHeap, Capacity);
+                Heap = NewHeap;
+                Capacity = NewCapacity;
+            }
+
+            private bool Dominates(T x, T y)
+            {
+                return Comparer.Compare(x, y) <= 0;
+            }
+
+        }
+
+        /// <summary>
+        /// Comparator which sorts Vbox instances based on their volume, in descending order
+        /// </summary>
+        private sealed class VBOX_COMPARATOR_VOLUME : Comparer<Vbox>
```

Comment:
Please do not use "_" at class name.

"VBOX_COMPARATOR_VOLUME" -> "VboxComparatorVolume"